.github/workflows/label-first-time-contributor.yml

```diff
@@ -0,0 +1,44 @@
+name: Label First-Time Contributor PRs
+
+on:
+  pull_request_target:
+    types:
+      - opened
+
+jobs:
+  triage:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Add 'new contributor' label
+        uses: actions/github-script@v6.3.3
+        with:
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          script: |
+            // Get a list of all issues created by the PR opener
+            const creator = context.payload.sender.login
+            const opts = github.rest.issues.listForRepo.endpoint.merge({
+              ...context.issue,
+              creator,
+              state: 'all'
+            })
+
+            const labelToAdd = 'new contributor';
+            const issues = await github.paginate(opts)
+
+            for (const issue of issues) {
+              if (issue.number === context.issue.number) {
+                continue
+              }
+
+              if (issue.pull_request) {
+                return // Creator is already a contributor.
```

Comment:
This only adds the new contributer label to the first PR of an author*, but the author can create multiple PRs before any of them is merged, in which case all of these PRs will require approval. I think the correct logic is to add the label if the author does not have any merged PR when a PR is created.

(* Or, strictly speaking, this only adds the label if it's the author's only PR when the workflow runs, so if the author creates another PR before the workflow runs, which can be several hours after the PR is created if github action is busy, none of the PRs will have the label added.)

It would also be nice if the label is removed after the workflows are approved, maybe by adding a new workflow that only runs with approval which removes the label.

Reply:
I'd need help sorting those things out

Reply:
You can refer to the [`octokit`](https://octokit.github.io/octokit.js/v18/) documention on how to list PRs and check their merged status, but after some thought I think it's very tricky to get it right because I don't know how exactly github requests/dismisses workflow approvals, and the workflows can also be started with a delay and in different orders than their creation order, which makes it even more difficult to analyze.

I also thought about using a dummy workflow that triggers on pull request creation and another workflow that triggers on the former workflow's creation/start using the [`workflow_run` event](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#workflow_run) to probe a PR's approval status, but it also turned out to be trickier than I had expected because the event data for the `workflow_run` event does not contain the exact id of the PR that triggered the dummy workflow, only the PRs that have the same head commit as the commit that triggered the dummy workflow, which may include other PRs.

Perhaps it'll be more sane if we just settle for a new contributor label, so we only need to solve the problem mentioned in the second paragraph of my previous comment, by checking that the author's other PRs are created after the PR triggering the workflow.